src/lib/components/workflow/filter-search/datetime-filter.svelte

```diff
@@ -239,18 +152,117 @@
             />
-            <div class="flex justify-end">
-              <Button variant="ghost" on:click={onApply}
-                >{translate('apply')}</Button
+          </div>
+        </MenuItem>
+        <MenuDivider />
+        <MenuItem>
+          <div class="flex flex-col gap-2">
+            <DatePicker
+              label={translate('end')}
+              on:datechange={onEndDateChange}
+              selected={endDate}
+              todayLabel={translate('today')}
+              closeLabel={translate('close')}
+              clearLabel={translate('clear-input-button-label')}
+            />
+            <TimePicker
+              bind:hour={endHour}
+              bind:minute={endMinute}
+              bind:second={endSecond}
+              twelveHourClock={false}
+            />
+          </div>
+        </MenuItem>
+      {:else}
+        <MenuItem on:click={() => (type = 'relative')}>
+          <div class="flex flex-col">
+            <label
+              class="flex flex-row items-center gap-2 cursor-pointer"
+              for="relative-time"
+            >
+              <input
+                on:click|stopPropagation={() => {
+                  type = 'relative';
+                }}
+                class="w-4 h-4 accent-gray-900"
+                type="radio"
+                checked={type === 'relative'}
+                id="relative-time"
+                tabindex="-1"
+              />
+              {translate('relative')}
+            </label>
+            <div class="ml-6 pt-2 flex gap-0">
+              <Input
+                label={translate('relative')}
+                labelHidden
+                id="relative-datetime-input"
+                bind:value={relativeTime}
+                placeholder="00"
+                error={error(relativeTime)}
+                unroundRight
+                class="h-10"
+              />
+              <Select
+                unroundLeft
+                bind:value={timeUnit}
+                id="relative-datetime-unit-input"
+                label={translate('time-unit')}
+                labelHidden
               >
+                {#each TIME_UNIT_OPTIONS as unit}
+                  <Option value={unit}>{unit} {translate('ago')}</Option>
+                {/each}
+              </Select>
+            </div>
+          </div>
+        </MenuItem>
+        <MenuDivider />
+        <MenuItem on:click={() => (type = 'absolute')}>
+          <div class="flex flex-col gap-2">
+            <label
+              class="flex flex-row items-center gap-2 cursor-pointer"
+              for="absolute-time"
+            >
+              <input
+                on:click|stopPropagation={() => (type = 'absolute')}
+                class="w-4 h-4 accent-gray-900"
+                type="radio"
+                checked={type === 'absolute'}
+                id="absolute-time"
+                tabindex="-1"
+              />
+              {translate('absolute')}
+            </label>
```

Comment:
I think it would be nice to extend `<Input />` to support `type="radio"`

Reply:
See https://github.com/temporalio/ui/pull/1613#discussion_r1317798158.